### PR TITLE
Jbruno/mgci1508 scheduler memory leak

### PR DESCRIPTION
**Pull request by JeanetteBruno**
- updated scheduler tests to validate consistency of its internal data structures 
- added reporting sizes of internal maps/slices to stats
- removed some unused code
- fixed some comments

This code has not yet been validated beyond unit testing

### CODE REVIEW

File: scheduler/server/load_based_sched_alg.go [hunk 1]
```diff
@@ -163,11 +163,6 @@ func (jc *jobClass) String() string {
 		jc.tempNormalizedPct)
 }
 
-// NewJobClass create a jobClass struct
```
Comment (JeanetteBruno) [inline scheduler/server/load_based_sched_alg.go:166]: this wasn't used anywhere

File: scheduler/server/load_based_sched_alg.go [hunk 2]
```diff
@@ -655,20 +650,20 @@ func (lbs *LoadBasedAlg) getNumTasksToStart(requestor string) int {
 // The function returns the list of tasks to stop and updates the jobClass objects with the number
 // of tasks to start
 func (lbs *LoadBasedAlg) rebalanceClassTasks(jobsByRequestor map[string][]*jobState, totalWorkers int) []*taskState {
-
+	log.Info("Rebalancing")
 	totalTasks := 0
 	// compute number tasks as per the each class's entitlement and waiting tasks
 	// will be negative when a class is over its entitlement
 	for _, jc := range lbs.jobClasses {
 		if jc.origNumRunningTasks > jc.origNumTargetedWorkers {
```
Comment (JeanetteBruno) [inline scheduler/server/load_based_sched_alg.go:658]: the comments were backwards

File: scheduler/server/job_state.go [hunk 3]
```diff
@@ -264,3 +270,38 @@ func (j *jobState) removeTaskFromStartTimeMap(jobID string, taskID string, start
 		delete(j.tasksByJobClassAndStartTimeSec, timeBucket)
 	}
 }
+
+func (j *jobState) logInconsistentStateValues() {
+	if j.TasksCompleted != len(j.Completed) {
+		log.Errorf("job:%s,%s,%s: TasksCompleted count (%d) != number entries in Completed map (%d)",
```
Comment (dgdonut) [inline scheduler/server/job_state.go]: suggest adding a magic string to the beginning of all these logs so it's easy to find any of them e.g. "inconsistent: "

File: scheduler/server/job_state.go [hunk 4]
```diff
@@ -264,3 +270,38 @@ func (j *jobState) removeTaskFromStartTimeMap(jobID string, taskID string, start
...
+	if j.TasksRunning != len(j.Running) {
+		log.Errorf("job:%s,%s,%s: TasksRunning count (%d) != number entries in Running map (%d)",
+			j.jobClass, j.Job.Def.Requestor, j.Job.Id, j.TasksRunning, len(j.Running))
+	}
+	notStarted := len(j.Tasks) - j.TasksRunning - j.TasksCompleted
+	if notStarted != len(j.NotStarted) {
+		log.Errorf("job:%s,%s,%s: TasksRunning count (%d) != number entries in Running map (%d)",
+			j.jobClass, j.Job.Def.Requestor, j.Job.Id, notStarted, len(j.NotStarted))
+	}
+	// TODO remove is slows things down too much
```
Comment (dgdonut) [inline scheduler/server/job_state.go]: typo: "remove if this slows..."
should we also add that we don't intend to release this past staging?

File: scheduler/server/stateful_scheduler.go [hunk 5]
```diff
@@ -636,6 +653,7 @@ func (s *statefulScheduler) checkJobsLoop() {
 						s.requestorHistory[rb] = []string{}
 					}
 					s.requestorHistory[rb] = append(s.requestorHistory[rb], checkJobMsg.jobDef.Tag)
+					s.requestorHistoryEntriesSize += int64(len(checkJobMsg.jobDef.Tag))
```
Comment (dgdonut) [inline scheduler/server/stateful_scheduler.go:651]: is this supposed to be adding the number of entries or the length of the Tag string?

**Review by dgdonut — APPROVED** [review]
Approving so we can debug, hopefully we can delete some or all of this later.

**Pull request merged**